### PR TITLE
BAH-3106 | Custom carbon components to be used for building Drug Chart Modal

### PR DESCRIPTION
**Pull request by Arjun-Go**
JIRA - https://bahmni.atlassian.net/jira/software/c/projects/BAH/boards/55?modal=detail&selectedIssue=BAH-3106

# Summary

This PR has changes to add custom carbon components aligned with Bahmni styles which can be leveraged while building new components. As part of this PR, components have been added for the following :-

- Dropdown
image

- TimePicker 12 Hour
image

- NumberInput
image

- DatePicker
image

- TimePicker 24 Hour
image

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,2 +1,9 @@
 import "./styles/styles.scss";
 export * from "./components/TestComp.jsx";
+export { default as DatePickerCarbon } from "./components/DatePickerCarbon/DatePickerCarbon.jsx";
```
Comment (bassoGeorge) [inline src/index.js:3]: You know, it is much easier to just have named exports everywhere. default exports in their files are a pain 😆

File: src/stories/TimePicker24Hour.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import moment from "moment";
+import TimePicker24Hour from "../components/TimePicker24Hour/TimePicker24Hour.jsx";
+
+export default {
+  title: "Time Picker 24 hour",
+};
+
+export const Primary = () => {
+  return (
+    <IntlProvider locale={"en"}>
```
Comment (bassoGeorge) [inline src/stories/TimePicker24Hour.stories.jsx]: See if you can make this <IntlProvider> wrapper common for all stories. You should be able to do it through the storybook mechanisms in preview.js I believe or else main.js

File: src/stories/DatePickerCarbon.stories.jsx
Comment (bassoGeorge) [inline src/stories/DatePickerCarbon.stories.jsx:1]: You know, let's colocate stories along with the component files. Having them in a separate stories/ folder just puts them out of mind

File: src/stories/util.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { IntlProvider } from "react-intl";
+import React from "react";
+
+export const withReactIntl = (Component, messages = {}) => {
```
Comment (bassoGeorge) [inline src/stories/util.js]: You have built this, but where is it used?

**Review by bassoGeorge — COMMENTED** [review]
I suggest colocating the stories.js files alongside the component files themselves

File: src/stories/DatePickerCarbon.stories.jsx
Comment (Arjun-Go) [reply src/stories/DatePickerCarbon.stories.jsx:1]: Sure. Makes sense

File: src/index.js (hunk 1 above)
Comment (Arjun-Go) [reply src/index.js:3]: I tried to do that like how you have done for the TestComp. But, named exports were not working surprisingly

File: src/stories/util.js (hunk 3 above)
Comment (Arjun-Go) [reply src/stories/util.js]: Removed this, it's not getting used anywhere

File: src/index.js (hunk 1 above)
Comment (bassoGeorge) [reply src/index.js:3]: that's weird... you sure your component was being exported as a named export too?

Comment (Arjun-Go) [reply src/index.js:3]: Yes, It was giving this error even though the exports were added
 
 ```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `Wr`.
Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `Wr`.
    at createFiberFromTypeAndProps (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:23965:21)
    at createFiberFromElement (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:23988:15)
    at reconcileSingleElement (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:14233:23)
    at reconcileChildFibers (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:14293:35)
    at reconcileChildren (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:16762:28)
    at updateHostComponent (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:17302:3)
    at beginWork (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:18627:14)
    at HTMLUnknownElement.callCallback (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:188:14)
    at Object.invokeGuardedCallbackDev (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:237:16)
    at invokeGuardedCallback (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:292:31)
```

File: src/stories/TimePicker24Hour.stories.jsx (hunk 2 above)
Comment (Arjun-Go) [reply src/stories/TimePicker24Hour.stories.jsx]: Done 👍

File: src/index.js (hunk 1 above)
Comment (Arjun-Go) [reply src/index.js:3]: Yup, added named exports in component and export like how it is done for TestComp in index.js. Still giving this error

```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `Wr`.
Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `Wr`.
    at createFiberFromTypeAndProps (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:23965:21)
    at createFiberFromElement (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:23988:15)
    at reconcileSingleElement (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:14233:23)
    at reconcileChildFibers (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:14293:35)
    at reconcileChildren (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:16762:28)
    at updateHostComponent (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:17302:3)
    at beginWork (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:18627:14)
    at HTMLUnknownElement.callCallback (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:188:14)
    at Object.invokeGuardedCallbackDev (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:237:16)
    at invokeGuardedCallback (webpack://openmrs-module-ipd-frontend/./node_modules/react-dom/cjs/react-dom.development.js?:292:31)

```

**Review by bassoGeorge — APPROVED** [review]
Looks good

**Pull request merged**